lib/containers/operations_docker.pm

```diff
@@ -0,0 +1,50 @@
+package containers::operations_docker;
+use base Exporter;
+use Exporter;
+use testapi;
+use version_utils qw(is_sle is_leap);
+use containers::utils qw(registry_url get_docker_version check_runtime_version container_ip);
+use Utils::Systemd 'systemctl';
+
+our @EXPORT = 'check_containers_firewall';
+
+sub check_containers_firewall {
+    record_info "firewall", "Checking that firewall is enabled, properly configured and containers can reach the Internet";
+    my $container_name = 'sut_container';
+    my $docker_version = get_docker_version();
+    systemctl('is-active firewalld');
+    my $running = script_output qq(docker ps -q | wc -l);
+    validate_script_output('ip a s docker0', sub { /state DOWN/ }) if $running == 0;
+    # Docker zone is created for docker version >= 20.10 (Tumbleweed), but it
+    # is backported to docker 19 for SLE15-SP3 and for Leap 15.3
+    if (check_runtime_version($docker_version, ">=20.10") || is_sle('>=15-sp3') || is_leap(">=15.3")) {
+        assert_script_run "firewall-cmd --list-all --zone=docker";
+        validate_script_output "firewall-cmd --list-interfaces --zone=docker", sub { /docker0/ };
+    }
+    # Rules applied before DOCKER. Default is to listen to all tcp connections
+    # ex. output: "1           0        0 RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0"
+    validate_script_output "iptables -L DOCKER-USER -nvx --line-numbers", sub { /1.+all.+0\.0\.0\.0\/0\s+0\.0\.0\.0\/0/ };
+
+    # Run container in the background
+    assert_script_run "docker run -id --rm --name $container_name -p 1234:1234 " . registry_url('alpine');
+    my $container_ip = container_ip($container_name, 'docker');
+
+    # Each running container should have added a new entry to the DOCKER zone.
+    # ex. output: "1           0        0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.2           tcp dpt:1234"
+    validate_script_output "iptables -L DOCKER -nvx --line-numbers", sub { /1.+ACCEPT.+!docker0 docker0.+$container_ip\s+tcp dpt:1234/ };
+
+    # Connectivity to host check
+    my $default_route = script_output "docker run " . registry_url('alpine') . " ip route show default | awk \'/default/ {print \$3}\'";
+    assert_script_run "docker run --rm " . registry_url('alpine') . " ping -c3 " . $default_route;
+
+    # Cross-container connectivity check
+    assert_script_run "docker run --rm " . registry_url('alpine') . " ping -c3 " . $container_ip;
+
+    # Outisde connectivity check
+    assert_script_run "docker run --rm " . registry_url('alpine') . " wget " . get_var('OPENQA_URL');
+
+    # Kill the container running on background
+    assert_script_run "docker kill $container_name ";
```

Comment:
Since everywhere the `--rm` flag is set, I don't believe this is needed?
But if it is a safety cleanup, why not make it `script_run` instead?

Reply:
No no, if you will go up, run run one container in the background. This needs to be done.